### DISCUSSION TITLE
Request for Documentation Update - Multi-GPU and Single Precision Support in MACE-LAMMPS

### DISCUSSION THREAD

**Discussion by clecust** [ACEsuit/mace#274] [General]
Hi,

I am reaching out to request enhancements to the documentation for your project, particularly in relation to multi-GPU and single precision support.

In the latest paper, "A foundation model for atomistic materials chemistry", the benefits of utilizing multiple GPUs are highlighted.     However, the existing documentation exclusively provides examples for single-GPU setups.     I have encountered an error while attempting to run the project on multiple GPUs, prompting the need for clearer guidance in the documentation.

I kindly request the addition of an introduction to multi-GPU usage in the documentation.     This would greatly assist users, like myself, who are interested in harnessing the advantages of multiple GPUs for large-scale simulations.

Additionally, it would be immensely helpful if an introduction to single precision support in mace-lammps could be incorporated into the documentation.

I appreciate your attention to these requests, as they are crucial for the successful implementation of my simulations.     
Thank you very much for your time and consideration.

**Comment by tawe141** [comment]
@wcwitt What is the state of single-precision models in LAMMPS at the moment? I'm happy to contribute if needed.

**Comment by wcwitt** [comment]
I believe it works and the only thing you need to change is [this line](https://github.com/ACEsuit/mace/blob/2d89108d4ec27ada6ae7e63b70f79c63ed1007e7/mace/cli/create_lammps_model.py#L14) in the script that creates the LAMMPS-wrapped model. If your model is already single-precision, you'd just need to delete the `.double()`.

Would you mind trying that to start?

**Reply by tawe141** [reply]
I tried it and the model returns nans in LAMMPS. Keeping `.double()` seems to work fine though. I've tried running the model in Python with single precision and it appears to work as well, so there might be something off with the LAMMPS interface specifically. Any guesses as to what that could be?

**Reply by wcwitt** [reply]
Was this with CPU or GPU? If GPU, was it with Kokkos? And how recently have you installed our LAMMPS fork?

**Reply by tawe141** [reply]
This is with GPU and KOKKOS. I pulled the LAMMPS fork maybe a couple weeks ago.

**Reply by wcwitt** [reply]
I will need to remind myself what was necessary to make things work with single-precision in Kokkos. If you're in a hurry, you could try [this branch](https://github.com/ACEsuit/lammps/tree/mace-kokkos-sp), which is probably the answer.

**Reply by wcwitt** [reply]
You could also try without Kokkos, where it should "just work". Do you have a GPU without Float64? Or are you just looking for a speedup?

**Reply by tawe141** [reply]
I'm just looking for a speedup. How do I run the model without Kokkos on the GPU?
